src/main/kotlin/model/Category.kt

```diff
@@ -0,0 +1,19 @@
+package model
+
+import java.io.Serializable
+import java.util.*
+
+private const val UNNAMED_CATEGORY = "UnnamedCategory"
+
+data class Category(
+    var id: Int,
+    var name: String = "",
+    var screenElements: MutableList<ScreenElement>
```

Comment:
There are some errors when you try to apply the settings. You need to initialize all properties here with some default values, right now there is no constructor without parameters and that's where serialization errors come from.

Reply:
Ah, yes. I'd been trying to figure out this issue for hours now. Thanks for that. 

One more issue is that the state gets persisted even when I'm pressing "Cancel".